### PR TITLE
Make docOutboundLog.POReference normal column

### PR DESCRIPTION
**Pull request by dm-ps**
refs: https://github.com/metasfresh/metasfresh/issues/18801

### CODE REVIEW

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments

File: backend/de.metas.document.archive/de.metas.document.archive.base/src/main/sql/postgresql/system/5732530_sys_gh18801_C_Doc_Outbound_Log_POReference.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+-- Column: C_Doc_Outbound_Log.POReference
+-- Column SQL (old): (SELECT getDocumentPOReference(C_Doc_Outbound_Log.AD_Table_ID, C_Doc_Outbound_Log.Record_ID))
+-- 2024-09-05T13:32:35.958Z
+UPDATE AD_Column SET ColumnSQL='', IsLazyLoading='N',Updated=TO_TIMESTAMP('2024-09-05 13:32:35.958000','YYYY-MM-DD HH24:MI:SS.US')::timestamp without time zone AT TIME ZONE 'UTC',UpdatedBy=100 WHERE AD_Column_ID=551964
+;
+
+-- 2024-09-05T13:32:42.395Z
+/* DDL */ SELECT public.db_alter_table('C_Doc_Outbound_Log','ALTER TABLE public.C_Doc_Outbound_Log ADD COLUMN POReference VARCHAR(40)')
+;
+
```
Comment (metas-ts) [inline backend/de.metas.document.archive/de.metas.document.archive.base/src/main/sql/postgresql/system/5732530_sys_gh18801_C_Doc_Outbound_Log_POReference.sql:12]: doing a DDL change and then setting the column in question in the same trx might leave to an error (the one with "pending trigger events").
better have a `commit;` here, or put the update into a dedicated file

Comment (dm-ps) [reply backend/de.metas.document.archive/de.metas.document.archive.base/src/main/sql/postgresql/system/5732530_sys_gh18801_C_Doc_Outbound_Log_POReference.sql]: separated in two files

**Review by metas-ts — APPROVED** [review]

File: backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/spi/impl/DocOutboundArchiveEventListener.java [hunk 2]
```diff
@@ -247,13 +254,35 @@ private I_C_Doc_Outbound_Log createLog(@NonNull final I_AD_Archive archiveRecord
 
 		setMailRecipient(docOutboundLogRecord);
 
+		setPOReference(docOutboundLogRecord, reference);
```
[the remark this answers is not in the document]
Comment (teosarca) [reply backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/spi/impl/DocOutboundArchiveEventListener.java:257]: @dm-ps @metas-ts disagree.
if you change the invoice's or inout's POReference then the model interceptors will kick in and not this BL.

[the remark this answers is not in the document]
Comment (teosarca) [reply backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/spi/impl/DocOutboundArchiveEventListener.java:257]: i don't see any benefit of this approach other than performance penalty

[the remark this answers is not in the document]
Comment (dm-ps) [reply backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/spi/impl/DocOutboundArchiveEventListener.java:257]: Will introduce `AD_Archive.POReference` and update both `AD_Archive.POReference` and `C_Doc_Outbound_Log.POReference` via interceptors

[the remark this answers is not in the document]
Comment (teosarca) [reply backend/de.metas.document.archive/de.metas.document.archive.base/src/main/java/de/metas/document/archive/spi/impl/DocOutboundArchiveEventListener.java:257]: thx

**Pull request merged**